src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -36,12 +38,34 @@
 
         :else (app req)))))
 
+(defn- blocking-submit-command [do-submit-fn response-pub uuid completion-timeout-ms]
+  (let [response-chan (async/chan)
+        topic (str uuid)
+        _ (async/sub response-pub topic response-chan)
+        timeout-chan (async/timeout completion-timeout-ms)
+        _ (do-submit-fn)
+        timed-out (async/alt!! timeout-chan true
+                               response-chan false)]
+    (async/unsub response-pub topic response-chan)
+    (http/json-response {:uuid uuid, :processed (not timed-out)}
+                        (if timed-out 503 200))))
+
```

Comment:
conclusion: you don't *have* to close a channel, but if you do you can prevent some otherwise unnecessary work.